### PR TITLE
Updated preview command to display new documentation.

### PR DESCRIPTION
**Pull request by freaz**

### CODE REVIEW

File: lib/apiary/command/preview.rb [hunk 1]
```diff
@@ -67,46 +75,12 @@ def rack_app(&block)
 
       def run_server
         app = self.rack_app do
-          self.query_apiary(@options.api_host, @options.path)
+          self.generate
```
Comment (abtris) [inline lib/apiary/command/preview.rb]: You can remove this `self`

File: lib/apiary/command/preview.rb [hunk 2]
```diff
@@ -116,17 +90,49 @@ def write_generated_path(path, outfile)
...
       end
 
-      def generate_static(path)
-        File.open(preview_path(path), "w") do |file|
-          file.write(query_apiary(@options.api_host, path))
+      def generate
+        template = load_preview_template()
+
+        data = {
+          title: File.basename(@options.path, ".*"),
```
Comment (abtris) [inline lib/apiary/command/preview.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: lib/apiary/command/preview.rb [hunk 3]
```diff
@@ -116,17 +90,49 @@ def write_generated_path(path, outfile)
...
+          blueprint: load_blueprint()
```
Comment (abtris) [inline lib/apiary/command/preview.rb]: Do not use parentheses for method calls with no arguments.

File: lib/apiary/file_templates/preview.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+</head>
+<body>
+  <script src="https://api.apiary.io/seeds/embed.js"></script>
+  <script>
+    var embed = new Apiary.Embed({
+      apiBlueprint: "<%= escape_javascript data[:blueprint] %>"
+    });
+  </script>
+</body>
+</html>
```
Comment (abtris) [inline lib/apiary/file_templates/preview.erb]: Add line end

File: lib/apiary/helpers/javascript_helper.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+          javascript.gsub(/(\\|<\/|\r\n|\342\200\250|\342\200\251|[\n\r"'])/u) {|match| JS_ESCAPE_MAP[match] }
+        else
+          ''
+        end
+      end
+
+      alias_method :j, :escape_javascript
+    end
+  end
+end
```
Comment (abtris) [inline lib/apiary/helpers/javascript_helper.rb]: Add line end

File: lib/apiary/command/preview.rb [hunk 6]
```diff
@@ -116,17 +90,49 @@ def write_generated_path(path, outfile)
...
         end
       end
 
+      def load_blueprint
+        file = File.open @options.path, "r"
+        file.read
+      end
+
+      def preview_path
+        basename = File.basename(@options.path, ".*")
```
Comment (abtris) [inline lib/apiary/command/preview.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: lib/apiary/command/preview.rb [hunk 7]
```diff
@@ -116,17 +90,49 @@ def write_generated_path(path, outfile)
...
+
+        File.open(preview_path, "w") do |file|
+          file.write preview_string
+          file.flush
           @options.output ? write_generated_path(file.path, @options.output) : open_generated_page(file.path)
         end
       end
 
+      def load_blueprint
+        file = File.open @options.path, "r"
```
Comment (abtris) [inline lib/apiary/command/preview.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: lib/apiary/command/preview.rb [hunk 8]
```diff
@@ -116,17 +90,49 @@ def write_generated_path(path, outfile)
...
+      end
+
+      def preview_path
+        basename = File.basename(@options.path, ".*")
+        temp = Dir.tmpdir()
+        "#{temp}/#{basename}-preview.html"
+      end
+
+      def load_preview_template
+        file = File.open(PREVIEW_TEMPLATE_PATH, "r")
```
Comment (abtris) [inline lib/apiary/command/preview.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: lib/apiary/command/preview.rb [hunk 9]
```diff
@@ -116,17 +90,49 @@ def write_generated_path(path, outfile)
...
+          blueprint: load_blueprint()
+        }
+
+        template.result(binding)
+      end
+
+      def generate_static
+        preview_string = generate
+
+        File.open(preview_path, "w") do |file|
```
Comment (abtris) [inline lib/apiary/command/preview.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

**Comment by abtris** [comment]
Looks good, only correct small issues. We have to add rubocop to apiary-client too as in redsnow.

**Comment by Almad** [comment]
@freaz Please always provide link to trello card

@abtris @freaz Has this been released and therefore can be communicated?

**Comment by honzajavorek** [comment]
- https://rubygems.org/gems/apiaryio is on 0.3.0, equals to the latest release here https://github.com/apiaryio/apiary-client/releases/tag/0.3.0
- already communicated here https://github.com/apiaryio/apiary-client/issues/39

**Pull request merged**